Add bbox support

mattficke opened the pull request:
Switches to using an options hash instead of another param. Fixes #7.

mattficke commented:
@tmcw @sbma44 since you two have contributed the bulk of the code here, can I get 👀 on this? Ok to merge?

sbma44 commented:
I was gonna suggest using Ruby's official URI building library but [apparently it doesn't have one](http://stackoverflow.com/questions/4654708/convert-ruby-hash-to-url-query-string-without-those-square-brackets). Oh well! As long as this passes tests I'm :+1:, though I think you probably could use some of the example code in that link to make it slightly more futureproof for additional arguments passed around in that hash.

tmcw commented on lib/mapbox/geocoder.rb:
```diff
@@ -6,17 +6,27 @@ class Geocoder
+        minY = bbox[1]
+        maxX = bbox[2]
+        maxY = bbox[3]
+        params += "#{params.length > 0 ? '&' : '?'}bbox=#{minX}%2C#{minY}%2C#{maxX}%2C#{maxY}"
```
maybe `bbox=#{bbox.join(',')}"` as a more concise version?

mattficke replied:
👍

tmcw commented on lib/mapbox/geocoder.rb:
```diff
@@ -6,17 +6,27 @@ class Geocoder
-    def self.geocode_forward(query, proximity=nil, dataset='mapbox.places')
-      proximity_param = ''
+    def self.geocode_forward(query, options={}, dataset='mapbox.places')
+      proximity = options[:proximity] || nil
```
since the default value of an unset property is `nil`, are these ||s only to save people who put false in their place?

mattficke replied:
Yes, I was trying to keep changes to a minimum and this avoided an undefined method error if `proximity = false`, but I can simplify things and just check `if proximity` instead.

mattficke commented:
Ok, refactored in response to your comments. I added a handler for extra params we may add in the future that don't need to be reformatted, and tests to illustrate that it works (for both real params like `country` and fake ones).

Since we have tests, I also added this to Circle CI.

Going to go ahead and merge if this looks ok. I think someone else will need to either publish this to rubygems, or give me publish permissions to do so.

sbma44 commented:
LGTM! I don't know who has the keys to rubygems, I'm afraid.

mattficke commented:
Merged. Looks like @tmcw last published to rubygems.
